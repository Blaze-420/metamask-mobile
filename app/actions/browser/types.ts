/**
 * Browser actions for Redux
 */
export enum BrowserActionTypes {
  ADD_TO_VIEWED_DAPP = 'ADD_TO_VIEWED_DAPP',
  SET_BROWSER_REQUEST_LOAD = 'SET_BROWSER_REQUEST_LOAD',
  SET_BROWSER_LOAD_START = 'SET_BROWSER_LOAD_START',
  SET_BROWSER_LOAD_END = 'SET_BROWSER_LOAD_END',
  ADD_TO_BROWSER_HISTORY = 'ADD_TO_BROWSER_HISTORY',
  CLEAR_BROWSER_HISTORY = 'CLEAR_BROWSER_HISTORY',
  ADD_TO_BROWSER_WHITELIST = 'ADD_TO_BROWSER_WHITELIST',
  CLOSE_ALL_TABS = 'CLOSE_ALL_TABS',
  CREATE_NEW_TAB = 'CREATE_NEW_TAB',
  CLOSE_TAB = 'CLOSE_TAB',
  SET_ACTIVE_TAB = 'SET_ACTIVE_TAB',
  UPDATE_TAB = 'UPDATE_TAB',
  STORE_FAVICON_URL = 'STORE_FAVICON_URL',
}

export type AddToViewedDappAction = {
  type: BrowserActionTypes.ADD_TO_VIEWED_DAPP;
  hostname: string;
};

export type BrowserRequestLoadAction = {
  type: BrowserActionTypes.SET_BROWSER_REQUEST_LOAD;
};

export type BrowserLoadStartAction = {
  type: BrowserActionTypes.SET_BROWSER_LOAD_START;
  payload: {
    url: string;
  };
};

export type BrowserLoadEndAction = {
  type: BrowserActionTypes.SET_BROWSER_LOAD_END;
  payload: {
    url: string;
  };
};

export type AddToHistoryAction = {
  type: BrowserActionTypes.ADD_TO_BROWSER_HISTORY;
  url: string;
  name: string;
};

export type ClearHistoryAction = {
  type: BrowserActionTypes.CLEAR_BROWSER_HISTORY;
  id: number;
};

export type AddToWhitelistAction = {
  type: BrowserActionTypes.ADD_TO_BROWSER_WHITELIST;
  url: string;
};

export type CloseAllTabsAction = {
  type: BrowserActionTypes.CLOSE_ALL_TABS;
};

export type CreateNewTabAction = {
  type: BrowserActionTypes.CREATE_NEW_TAB;
  url: string;
  id: number;
  linkType: string;
};

export type CloseTabAction = {
  type: BrowserActionTypes.CLOSE_TAB;
  id: string;
};

export type SetActiveTabAction = {
  type: BrowserActionTypes.SET_ACTIVE_TAB;
  id: string;
};

export type UpdateTabAction = {
  type: BrowserActionTypes.UPDATE_TAB;
  id: string;
  data: {
    url: string;
    image?: string;
  };
};

export type StoreFaviconAction = {
  type: BrowserActionTypes.STORE_FAVICON_URL;
  origin: string;
  url: string;
};

export type BrowserAction =
  | AddToViewedDappAction
  | BrowserRequestLoadAction
  | BrowserLoadStartAction
  | BrowserLoadEndAction
  | AddToHistoryAction
  | ClearHistoryAction
  | AddToWhitelistAction
  | CloseAllTabsAction
  | CreateNewTabAction
  | CloseTabAction
  | SetActiveTabAction
  | UpdateTabAction
  | StoreFaviconAction;
