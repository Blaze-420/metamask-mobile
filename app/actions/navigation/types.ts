import { type Action } from 'redux';

/**
 * Navigation action type enum
 */
export enum NavigationActionType {
  ON_NAVIGATION_READY = 'ON_NAVIGATION_READY',
}

export type OnNavigationReadyAction =
  Action<NavigationActionType.ON_NAVIGATION_READY>;

/**
 * Navigation action
 */
export type NavigationAction = OnNavigationReadyAction;
