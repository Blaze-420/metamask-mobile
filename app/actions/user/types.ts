import { AppThemeKey } from '../../util/theme/models';

// Action type enum
export enum UserActionType {
  LOCKED_APP = 'LOCKED_APP',
  AUTH_SUCCESS = 'AUTH_SUCCESS',
  AUTH_ERROR = 'AUTH_ERROR',
  INTERRUPT_BIOMETRICS = 'INTERRUPT_BIOMETRICS',
  LOGIN = 'LOGIN',
  LOGOUT = 'LOGOUT',
  ON_PERSISTED_DATA_LOADED = 'ON_PERSISTED_DATA_LOADED',
  PASSWORD_SET = 'PASSWORD_SET',
  PASSWORD_UNSET = 'PASSWORD_UNSET',
  SEEDPHRASE_BACKED_UP = 'SEEDPHRASE_BACKED_UP',
  SEEDPHRASE_NOT_BACKED_UP = 'SEEDPHRASE_NOT_BACKED_UP',
  BACK_UP_SEEDPHRASE_VISIBLE = 'BACK_UP_SEEDPHRASE_VISIBLE',
  BACK_UP_SEEDPHRASE_NOT_VISIBLE = 'BACK_UP_SEEDPHRASE_NOT_VISIBLE',
  PROTECT_MODAL_VISIBLE = 'PROTECT_MODAL_VISIBLE',
  PROTECT_MODAL_NOT_VISIBLE = 'PROTECT_MODAL_NOT_VISIBLE',
  LOADING_SET = 'LOADING_SET',
  LOADING_UNSET = 'LOADING_UNSET',
  SET_GAS_EDUCATION_CAROUSEL_SEEN = 'SET_GAS_EDUCATION_CAROUSEL_SEEN',
  SET_APP_THEME = 'SET_APP_THEME',
  CHECKED_AUTH = 'CHECKED_AUTH',
}

export type UserAction =
  | { type: UserActionType.LOCKED_APP }
  | {
      type: UserActionType.AUTH_SUCCESS;
      payload: { bioStateMachineId: string };
    }
  | { type: UserActionType.AUTH_ERROR; payload: { bioStateMachineId: string } }
  | { type: UserActionType.INTERRUPT_BIOMETRICS }
  | { type: UserActionType.LOGIN }
  | { type: UserActionType.LOGOUT }
  | { type: UserActionType.ON_PERSISTED_DATA_LOADED }
  | { type: UserActionType.PASSWORD_SET }
  | { type: UserActionType.PASSWORD_UNSET }
  | { type: UserActionType.SEEDPHRASE_BACKED_UP }
  | { type: UserActionType.SEEDPHRASE_NOT_BACKED_UP }
  | { type: UserActionType.BACK_UP_SEEDPHRASE_VISIBLE }
  | { type: UserActionType.BACK_UP_SEEDPHRASE_NOT_VISIBLE }
  | { type: UserActionType.PROTECT_MODAL_VISIBLE }
  | { type: UserActionType.PROTECT_MODAL_NOT_VISIBLE }
  | { type: UserActionType.LOADING_SET; loadingMsg: string }
  | { type: UserActionType.LOADING_UNSET }
  | { type: UserActionType.SET_GAS_EDUCATION_CAROUSEL_SEEN }
  | { type: UserActionType.SET_APP_THEME; payload: { theme: AppThemeKey } }
  | { type: UserActionType.CHECKED_AUTH; payload: { initialScreen: string } };
