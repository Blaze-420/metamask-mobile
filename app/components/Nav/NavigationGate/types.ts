/**
 * Props for the NavigationGate component
 */
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type NavigationGateProps = {
  children: React.ReactNode;
};
