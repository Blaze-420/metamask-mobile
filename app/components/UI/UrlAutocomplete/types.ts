/**
 * Props for the UrlAutocomplete component
 */

import { Hex } from '@metamask/utils';

// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type UrlAutocompleteComponentProps = {
  /**
   * Callback that is triggered while
   * choosing one of the autocomplete options
   */
  onSelect: (item: AutocompleteSearchResult) => void;
  /**
   * Callback that is triggered while
   * tapping on the background
   */
  onDismiss: () => void;
};

/**
 * Ref for the UrlAutocomplete component
 */
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type UrlAutocompleteRef = {
  /**
   * Search for autocomplete results
   */
  search: (text: string) => void;
  /**
   * Hide the autocomplete results
   */
  hide: () => void;
  /**
   * Show the autocomplete results
   */
  show: () => void;
};

/**
 * The result of a Fuse search
 */
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type FuseSearchResult = {
  type: 'sites' | 'recents' | 'favorites';
  url: string;
  name: string;
};

/**
 * The result of a token search
 */
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type TokenSearchResult = {
  type: 'tokens';
  name: string;
  symbol: string;
  address: string;
  chainId: Hex;
  logoUrl?: string;
};

export type AutocompleteSearchResult = FuseSearchResult | TokenSearchResult;
