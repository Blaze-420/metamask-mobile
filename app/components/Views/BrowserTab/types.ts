export interface IpfsContentResult {
  url?: string;
  hash?: string;
  type?: string;
  reload?: boolean;
}

export type SessionENSNames = {
  [key: string]: {
    hostname: string;
    hash: string;
    type: string;
  };
};

/**
 * The props for the BrowserTab component
 */
export type BrowserTabProps = {
  /**
   * The ID of the current tab
   */
  id: number;
  /**
   * The ID of the active tab
   */
  activeTab: number;
  /**
   * InitialUrl
   */
  initialUrl?: string;
  /**
   * linkType - type of link to open
   */
  linkType?: string;
  /**
   * Protocol string to append to URLs that have none
   */
  defaultProtocol: string;
  /**
   * A string that of the chosen ipfs gateway
   */
  ipfsGateway: string;
  /**
   * Object containing the information for the current transaction
   */
  transaction?: Record<string, unknown>;
  /**
   * react-navigation object used to switch between screens
   */
  navigation: any; // Consider using proper react-navigation types
  /**
   * A string that represents the selected address
   */
  selectedAddress: string;
  /**
   * whitelisted url to bypass the phishing detection
   */
  whitelist: string[];
  /**
   * Url coming from an external source
   * For ex. deeplinks
   */
  url?: string;
  /**
   * Function to open a new tab
   */
  newTab: (url?: string) => void;
  /**
   * Function to store bookmarks
   */
  addBookmark: (bookmark: { name: string; url: string }) => void;
  /**
   * Array of bookmarks
   */
  bookmarks: Array<{ name: string; url: string }>;
  /**
   * String representing the current search engine
   */
  searchEngine: string;
  /**
   * Function to store the a page in the browser history
   */
  addToBrowserHistory: (entry: { url: string; name: string }) => void;
  /**
   * Function to store the a website in the browser whitelist
   */
  addToWhitelist: (url: string) => void;
  /**
   * Function to update the tab information
   */
  updateTabInfo: (url: string, tabID: number) => void;
  /**
   * Function to update the tab information
   */
  showTabs: () => void;
  /**
   * Action to set onboarding wizard step
   */
  setOnboardingWizardStep: (step: number) => void;
  /**
   * Current onboarding wizard step
   */
  wizardStep: number;
  /**
   * the current version of the app
   */
  app_version?: string;
  /**
   * Represents ipfs gateway toggle
   */
  isIpfsGatewayEnabled: boolean;
  /**
   * Represents the current chain id
   */
  chainId: string;
  /**
   * Boolean indicating if browser is in tabs view
   */
  isInTabsView: boolean;
};
