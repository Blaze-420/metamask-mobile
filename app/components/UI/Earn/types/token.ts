import { TokenI as BaseTokenI } from '../../Tokens/types';

export interface EarnTokenI extends BaseTokenI {
  tokenBalanceFormatted: string;
  balanceFiat: string;
}
