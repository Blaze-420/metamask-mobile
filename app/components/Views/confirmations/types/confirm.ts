import { ReactElement } from 'react';

interface Alert {
  alertDetails?: string[];
  component?: ReactElement;
  cta?: { label: string; callback: () => void };
  field?: string;
  isBlocking?: boolean;
  key: string;
  message: string;
  severity: 'danger' | 'warning';
  title: string;
}

export default Alert;
