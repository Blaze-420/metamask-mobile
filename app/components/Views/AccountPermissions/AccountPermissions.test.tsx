import React from 'react';
import renderWithProvider, {
  DeepPartial,
} from '../../../util/test/renderWithProvider';
import { backgroundState } from '../../../util/test/initial-root-state';
import { RootState } from '../../../reducers';
import AccountPermissions from './AccountPermissions';
import { ConnectedAccountsSelectorsIDs } from '../../../../e2e/selectors/Browser/ConnectedAccountModal.selectors';
import { fireEvent } from '@testing-library/react-native';
import { AccountPermissionsScreens } from './AccountPermissions.types';
import { updatePermittedChains, addPermittedAccounts } from '../../../core/Permissions';
import { NetworkConnectMultiSelectorSelectorsIDs } from '../../../../e2e/selectors/Browser/NetworkConnectMultiSelector.selectors';
import { ConnectAccountBottomSheetSelectorsIDs } from '../../../../e2e/selectors/Browser/ConnectAccountBottomSheet.selectors';

const mockedNavigate = jest.fn();
const mockedGoBack = jest.fn();
const mockedTrackEvent = jest.fn();

jest.mock('../../../core/Engine', () => ({
  context: {
    NetworkController: {
      setActiveNetwork: jest.fn(),
    },
    PermissionController: {
      revokeAllPermissions: jest.fn(),
    },
    KeyringController: {
      state: {
        keyrings: [],
      }
    },
    AccountsController: {
      state: {
        internalAccounts: {
          accounts: {}
        }
      }
    }
  },
}));

jest.mock('../../../core/Permissions', () => ({
  ...jest.requireActual('../../../core/Permissions'),
  updatePermittedChains: jest.fn(),
  addPermittedAccounts: jest.fn(),
}));
const mockUpdatePermittedChains = updatePermittedChains as jest.Mock;
const mockAddPermittedAccounts = addPermittedAccounts as jest.Mock;

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
      goBack: mockedGoBack,
    }),
  };
});

jest.mock('../../../components/hooks/useMetrics', () => ({
  useMetrics: () => ({
    trackEvent: mockedTrackEvent,
  }),
}));

jest.mock('react-native-safe-area-context', () => {
  const inset = { top: 0, right: 0, bottom: 0, left: 0 };
  const frame = { width: 0, height: 0, x: 0, y: 0 };
  return {
    SafeAreaProvider: jest.fn().mockImplementation(({ children }) => children),
    SafeAreaConsumer: jest
      .fn()
      .mockImplementation(({ children }) => children(inset)),
    useSafeAreaInsets: jest.fn().mockImplementation(() => inset),
    useSafeAreaFrame: jest.fn().mockImplementation(() => frame),
  };
});

jest.mock('../../hooks/useAccounts', () => {
  const useAccountsMock = jest.fn(() => ({
    evmAccounts: [
      {
        name: 'Account 1',
        address: '0xC4955C0d639D99699Bfd7Ec54d9FaFEe40e4D272',
        assets: {
          fiatBalance: '$3200.00\n1 ETH',
          tokens: [],
        },
        type: 'HD Key Tree',
        yOffset: 0,
        isSelected: true,
        balanceError: undefined,
      },
      {
        name: 'Account 2',
        address: '0xd018538C87232FF95acbCe4870629b75640a78E7',
        assets: {
          fiatBalance: '$6400.00\n2 ETH',
          tokens: [],
        },
        type: 'HD Key Tree',
        yOffset: 78,
        isSelected: false,
        balanceError: undefined,
      },
    ],
    accounts: [],
    ensByAccountAddress: {},
  }));
  return {
    useAccounts: useAccountsMock,
    Account: Object, // Mock for the Account type
  };
});

const mockInitialState: DeepPartial<RootState> = {
  settings: {},
  engine: {
    backgroundState: {
      ...backgroundState,
    },
  },
};

describe('AccountPermissions', () => {
  it('renders correctly', () => {
    const { toJSON } = renderWithProvider(
      <AccountPermissions
        route={{
          params: {
            hostInfo: { metadata: { origin: 'test' } },
          },
        }}
      />,
      { state: mockInitialState },
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should handle manage permissions button press and navigate to permissions summary', () => {
    const { getByTestId } = renderWithProvider(
      <AccountPermissions
        route={{
          params: {
            hostInfo: { metadata: { origin: 'test' } },
          },
        }}
      />,
      { state: mockInitialState },
    );

    const managePermissionsButton = getByTestId(
      ConnectedAccountsSelectorsIDs.MANAGE_PERMISSIONS,
    );
    fireEvent.press(managePermissionsButton);

    expect(getByTestId('permission-summary-container')).toBeDefined();
  });

  it('should render connect more accounts screen when specified as initial screen', () => {
    const { getByText } = renderWithProvider(
      <AccountPermissions
        route={{
          params: {
            hostInfo: { metadata: { origin: 'test' } },
            initialScreen: AccountPermissionsScreens.ConnectMoreAccounts,
          },
        }}
      />,
      { state: mockInitialState },
    );

    expect(getByText('Connect more accounts')).toBeDefined();
  });

  it('handles update permissions when networks are selected', async () => {
    const { getByText, getByTestId } = renderWithProvider(
      <AccountPermissions
        route={{
          params: {
            hostInfo: { metadata: { origin: 'test' } },
            initialScreen: AccountPermissionsScreens.ConnectMoreNetworks,
          },
        }}
      />,
      { state: mockInitialState },
    );

    // Select a network
    const network = getByText('Ethereum Mainnet');
    fireEvent.press(network);

    // Press update button
    const updateButton = getByTestId(
      NetworkConnectMultiSelectorSelectorsIDs.UPDATE_CHAIN_PERMISSIONS,
    );
    fireEvent.press(updateButton);

    expect(mockUpdatePermittedChains).toHaveBeenCalledWith('test', [
      '0x1'
    ], true);
  });

  it('handles the revoke permissions modal when no networks are selected', async () => {
    const { getByTestId } = renderWithProvider(
      <AccountPermissions
        route={{
          params: {
            hostInfo: { metadata: { origin: 'test' } },
            initialScreen: AccountPermissionsScreens.ConnectMoreNetworks,
          },
        }}
      />,
      { state: mockInitialState },
    );

    // Press revoke button
    const revokeButton = getByTestId(
      ConnectedAccountsSelectorsIDs.DISCONNECT_NETWORKS_BUTTON,
    );
    fireEvent.press(revokeButton);

    expect(mockedNavigate).toHaveBeenCalled();
  });

  it('handles update permissions when accounts are selected', async () => {
    const { getByText, getByTestId } = renderWithProvider(
      <AccountPermissions
        route={{
          params: {
            hostInfo: { metadata: { origin: 'test' } },
            initialScreen: AccountPermissionsScreens.EditAccountsPermissions,
          },
        }}
      />,
      { state: mockInitialState },
    );

    // Select an account
    const account = getByText('Account 1');
    fireEvent.press(account);

    // Press update button
    const updateButton = getByTestId(
      ConnectAccountBottomSheetSelectorsIDs.SELECT_MULTI_BUTTON,
    );
    fireEvent.press(updateButton);

    expect(mockAddPermittedAccounts).toHaveBeenCalledWith('test', [
      '0xC4955C0d639D99699Bfd7Ec54d9FaFEe40e4D272'
    ]);
  });

  it('handles the revoke permissions modal when no accounts are selected', async () => {
    const { getByTestId } = renderWithProvider(
      <AccountPermissions
        route={{
          params: {
            hostInfo: { metadata: { origin: 'test' } },
            initialScreen: AccountPermissionsScreens.EditAccountsPermissions,
          },
        }}
      />,
      { state: mockInitialState },
    );

    // Press revoke button
    const revokeButton = getByTestId(
      ConnectedAccountsSelectorsIDs.DISCONNECT_NETWORKS_BUTTON,
    );
    fireEvent.press(revokeButton);

    expect(mockedNavigate).toHaveBeenCalled();
  });
});
