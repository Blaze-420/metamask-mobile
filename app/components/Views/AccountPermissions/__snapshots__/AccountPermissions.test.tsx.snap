// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPermissions renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#00000099",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "opacity": 0,
        },
      ]
    }
  >
    <TouchableOpacity
      onPress={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "backgroundColor": "#ffffff",
            "borderColor": "#bbc0c566",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 1,
            "maxHeight": 1334,
            "overflow": "hidden",
            "paddingBottom": 0,
            "shadowColor": "#0000001a",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 40,
          },
          {
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "padding": 4,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#bbc0c566",
              "borderRadius": 2,
              "height": 4,
              "width": 40,
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 16,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderRadius": 16,
              "height": 16,
              "marginRight": 8,
              "overflow": "hidden",
              "width": 16,
            }
          }
        >
          <Image
            onError={[Function]}
            resizeMode="contain"
            source={{}}
            style={
              {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
            testID="favicon-avatar-image"
          />
        </View>
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#141618",
              "fontFamily": "EuclidCircularB-Bold",
              "fontSize": 18,
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          test
        </Text>
      </View>
      <View
        style={
          {
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#141618",
              "fontFamily": "EuclidCircularB-Medium",
              "fontSize": 14,
              "fontWeight": "500",
              "letterSpacing": 0,
              "lineHeight": 22,
              "marginVertical": 16,
            }
          }
          testID="accounts-connected-network-picker"
        >
          <View
            style={null}
          >
            <View
              accessibilityLabel="Ethereum Main Network"
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "width": 16,
                }
              }
              testID="network-avatar-picker"
            >
              <Image
                onError={[Function]}
                resizeMode="contain"
                source={1}
                style={
                  {
                    "height": 16,
                    "width": 16,
                  }
                }
                testID="network-avatar-image"
              />
            </View>
          </View>
          <Text
            accessibilityRole="text"
            numberOfLines={1}
            style={
              {
                "color": "#141618",
                "flexShrink": 1,
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 22,
                "marginHorizontal": 8,
              }
            }
            testID="open-networks-text"
          >
            Ethereum Main Network
          </Text>
          <SvgMock
            color="#141618"
            height={12}
            name="ArrowDown"
            style={
              {
                "height": 12,
                "width": 12,
              }
            }
            testID="pickernetwork-arrow"
            width={12}
          />
        </TouchableOpacity>
      </View>
      <RCTScrollView
        collapsable={false}
        data={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        initialNumToRender={999}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        renderScrollComponent={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View />
      </RCTScrollView>
      <View
        style={
          {
            "marginVertical": 16,
          }
        }
        testID="connect-accounts-buttons"
      >
        <View>
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={1}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderRadius": 24,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#0376c9",
                  "fontFamily": "EuclidCircularB-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Connect more accounts
            </Text>
          </TouchableOpacity>
        </View>
      </View>
      <TouchableOpacity
        accessibilityRole="button"
        accessible={true}
        activeOpacity={1}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "borderColor": "#0376c9",
            "borderRadius": 24,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingHorizontal": 16,
          }
        }
        testID="manage_permissions"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#0376c9",
              "fontFamily": "EuclidCircularB-Medium",
              "fontSize": 14,
              "fontWeight": "500",
              "letterSpacing": 0,
              "lineHeight": 22,
            }
          }
        >
          Manage Permissions
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
